core/bug_api.php

```diff
@@ -985,7 +985,14 @@ function bug_text_clear_cache( $p_bug_id = null ) {
  * @access public
  */
 function bug_exists( $p_bug_id ) {
-	if( false == bug_cache_row( $p_bug_id, false ) ) {
+	$c_bug_id = (int)$p_bug_id;
+
+	# Prevent db query errors
+	if( $c_bug_id > db_max_uint() ) {
+		return false;
+	}
+
+	if( false == bug_cache_row( $c_bug_id, false ) ) {
```

Comment:
Not necessarily.
== will match for equivalent values like 0, '', null... etc, Even if the function returned any of those, the condition should be still be matched, since no real bug data was found.
However, "false ==" is misleading, as "false" is not the only value we would want to match.
In this case i prefer the syntax (here inversed, though)

    if( bug_cache_row( $c_bug_id, false ) )